Initial Plugin

byronic opened the pull request:
This PR contains the initial work on the (eventual) 0.1.0 release of this plugin.

As of the time of this writing, executing this plugin during a mvn site run correctly generates a table of contents + report set on the Audits page it adds to the maven site.

byronic committed:
Initial Version

First working version; essentially no tests

byronic committed:
Remove bug hunt logging

byronic committed:
Add examples

byronic committed:
Parallel Stream bugfix, Add Metrics

Adds missing metrics to the site report; changes some calls during report generation from parallelstream to stream since they are manipulating the same Sink

Commit:
Parse Markdown, Update Dependencies, Tests

Now parses Markdown via the doxia markdown module for the introduction and confidentiality statement sections; updates parent and dependencies appropriately to reduce drift; adds some testing

Commit:
Use released auditor-api, up required test coverage

mykelalvis commented on examples/example-reporter-impl/pom.xml:
```diff
@@ -0,0 +1,56 @@
+  <parent>
+    <groupId>org.infrastructurebuilder</groupId>
+    <artifactId>ibparent</artifactId>
+    <version>24</version>
```
28

mykelalvis commented on pom.xml:
```diff
@@ -0,0 +1,97 @@
+  <parent>
+    <groupId>org.infrastructurebuilder</groupId>
+    <artifactId>ibparent</artifactId>
+    <version>26</version>
```
28

mykelalvis commented on src/main/java/org/infrastructurebuilder/auditor/AuditorReportMojo.java:
```diff
@@ -0,0 +1,83 @@
+  /**
+   * Where the HTML pages of the report will be created
+   */
+  @Parameter(defaultValue = "${project.reporting.outputDirectory}", property = "outputDirectory", required = true)
```
prolly `property = "audit.outputDirectory"`

mykelalvis commented on src/main/java/org/infrastructurebuilder/auditor/AuditorReportMojo.java:
```diff
@@ -0,0 +1,83 @@
+  private PageGenerator pageGenerator;
+
+  public void setPageGenerator(PageGenerator pageGenerator) {
+    this.pageGenerator = Objects.requireNonNull(pageGenerator);
```
`import static java.util.Objects.requireNonNull`

byronic committed:
Remove helpmojo, Change Test Percentage

mykelalvis commented on src/main/java/org/infrastructurebuilder/auditor/PageGenerator.java:
```diff
@@ -0,0 +1,166 @@
+
+    long duration = Duration.between(results.getTimestampStart().toInstant(), results.getTimestampEnd().toInstant())
+        .toMillis();
+    resultMetrics(results.getDescriptionHeaders(), results.getResults(), duration, sink);
```
`Duration` is an available type.  Mebbe `AuditResults` has a `Duration getDuration()`?

byronic replied:
Counterpoint:

```
Caused by: org.codehaus.modello.model.ModelValidationException: Field 'duration': Illegal type: 'java.time.Duration'.
    at org.codehaus.modello.model.ModelField.validateElement (ModelField.java:224)
```

mykelalvis replied:
Sorry.  `Duration` is an available *Java* type.

mykelalvis commented on src/test/java/org/infrastructurebuilder/auditor/TestAuditorReportMojo.java:
```diff
@@ -0,0 +1,28 @@
+  @Test
+  public void testConstructor() {
+    AuditorReportMojo m = new AuditorReportMojo();
+    assertEquals("audits", m.getOutputName());
```
Set the setters, yo.

mykelalvis commented on examples/example-audit-execution/pom.xml:
```diff
@@ -0,0 +1,71 @@
+<!--
```
This should probably be in `src/it/example` and executed by invoker (somewhere in the tree).

byronic committed:
Updates To API, Split Into Test/Site Packages

Updates examples, auditor API, and refactors the execution of the audits to be separate from the site build

byronic committed:
Rename outputDirectory to audit.outputDirectory

byronic committed:
Integration Tests

Moves examples to integration test directory

byronic committed:
Add package mojo

Adds package mojo, corrects tests to include it

byronic committed:
Update Dependency Versions, Correct Integration

Updates parent and auditor-api dependencies to latest releases; corrects integration test poms to use filtered project version

mykelalvis commented on pom.xml:
```diff
@@ -20,23 +20,35 @@
+      <dependency>
+        <groupId>org.infrastructurebuilder.util</groupId>
+        <artifactId>ibcore-bom</artifactId>
+        <version>0.15.0.0</version>
```
0.16.0.1 is out....

mykelalvis reviewed (changes requested):
No requirement for moving to the new core, but recommended.

mykelalvis commented:
The build fails.  Looks like the invoker is being invoked before install happens?  Maybe the invoker plugin got an upgrade, because I see some new goals available in it and I didn't really keep up with it for the last few months.  Let's take a look at the goals other than `invoker:run` .

mykelalvis committed:
Patch to make invoker run

mykelalvis commented:
I added `invoker:install` before `invoker:run` as a test and it gets farther, but you can't have one integration test depend on another unless you produce ordering of the names themselves.  (`example-audit-execution` happens before it's dependency `example-reporter-impl` is built and installed).

I created #2 to denote the changes that fix this.

byronic committed:
Merge pull request #2 from infrastructurebuilder/feature/initial-updates

Patch to make invoker run

byronic committed:
Update ibcore-bom version

byronic commented:
I'll bet I was getting a false positive from the local repository level re: dependencies. Let me clean local and rebuild, see if anything else shakes out

mykelalvis commented:
Ship it
